### PR TITLE
Ticket 741

### PR DESCRIPTION
**Pull request by lewisodriscoll**
Builds on pull request #89 to fix ticket [#741](http://trac.sasview.org/ticket/741)

- P(Q) and S(Q) are now labelled correctly (the labels were previously swapped)
- Any plots of P(Q) and S(Q) are now updated when parameter values are changed

Note: This requires the sasmodels branch ticket-741 in order to run

### CODE REVIEW

File: src/sas/sasgui/perspectives/fitting/fitting.py [hunk 1]
```diff
@@ -1756,14 +1756,38 @@ def _complete1D(self, x, y, page_id, elapsed, index, model,
                                       data_description="Data unsmeared",
                                       data_id="Data  " + data.name + " unsmeared",
                                       dy=unsmeared_error)
-        # Comment this out until we can get P*S models with correctly populated parameters
         if sq_model is not None and pq_model is not None:
-            self.create_theory_1D(x, sq_model, page_id, model, data, state,
+            # Store which data sets are plotted in which windows
+            # plot_ids[data_id] = [group_id1, group_id2, ...]
+            plot_ids = {}
+            for plot_panel in self.parent.plot_panels.values():
```
Comment (lewisodriscoll) [inline src/sas/sasgui/perspectives/fitting/fitting.py]: This seems a bit hacky, is there a better way to check if any plots of the theories exist?

**Comment by RichardHeenan** [comment]
P(Q) & S(Q) labels still seem swapped, if I did the check-outs correctly ....
Changes to P(Q) and S(Q) appear immediately when they have been sent to a new plot, but not if they have been "appended" to a plot. There are some other quirky behaviours re what graphs appear when and how they update, but they have I think always been there. We may need to give users specific instructions on what to do. Will try to discuss in person with Lewis.

Richard

**Comment by lewisodriscoll** [comment]
All issues RE: @RichardHeenan's comments should be fixed now.

I've noticed that if you produce a new plot of your data after sending it to the fitting perspective, close this new plot, then click fit, the updated model curve is plotted in a new window, instead of updating the original one. This behaviour is present in the 4.1.2 release so doesn't seem related to the P(Q) & S(Q) plots; should I open a new ticket?

**Comment by RichardHeenan** [comment]
I suppose you should, I have seen similar or same myself.
Will check your new code when I'm in next week...
Thanks
Richard

Sent from my Windows 10 phone

**Comment by lewisodriscoll** [comment]
I've raised this as [ticket #991](http://trac.sasview.org/ticket/991)

**Comment by butlerpd** [comment]
Richard says he will check once he is back in the office?  I will wait thus wait to merge till he does so.  In the meantime, I will try to do some functional testing myself this week.

**Comment by RichardHeenan** [comment]
I’ve been doing ESS stuff all today (Andrew will be happy), and am going away again,  not back in until 11 or 12 Sept, so if anyone else has time to test this please go ahead. You may still have to pull Lewis’ fork rather than a branch …
Richard

**Pull request merged**